--- a/jpeg/jconfig.h
+++ b/jpeg/jconfig.h
@@ -1,3 +1,8 @@
+/* FLTK *************************************************************** */
+/* FLTK   Comments marked with FLTK show modifications for FLTK which   */
+/* FLTK   should probably be preserved when the JPEG lib is upgraded.   */
+/* FLTK *************************************************************** */
+
 /* jconfig.h.  Generated from jconfig.cfg by configure.  */
 /* jconfig.cfg --- source file edited by configure script */
 /* see jconfig.txt for explanations */
@@ -8,9 +13,9 @@
 /* #undef void */
 /* #undef const */
 /* #undef CHAR_IS_UNSIGNED */
-#define HAVE_STDDEF_H 1
+/* #define HAVE_STDDEF_H 1 */		/* FLTK */
 #define HAVE_STDLIB_H 1
-#define HAVE_LOCALE_H 1
+/* #define HAVE_LOCALE_H 1 */		/* FLTK */
 /* #undef NEED_BSD_STRINGS */
 /* #undef NEED_SYS_TYPES_H */
 /* #undef NEED_FAR_POINTERS */
@@ -35,7 +40,7 @@ typedef unsigned char boolean;
 #ifdef JPEG_INTERNALS
 
 /* #undef RIGHT_SHIFT_IS_UNSIGNED */
-#define INLINE __inline__
+/* #define INLINE __inline__ */		/* FLTK */
 /* These are for configuring the JPEG memory manager. */
 /* #undef DEFAULT_MAX_MEM */
 /* #undef NO_MKTEMP */
--- a/jpeg/jpeglib.h
+++ b/jpeg/jpeglib.h
@@ -1,3 +1,8 @@
+/* FLTK *************************************************************** */
+/* FLTK   Comments marked with FLTK show modifications for FLTK which   */
+/* FLTK   should probably be preserved when the JPEG lib is upgraded.   */
+/* FLTK *************************************************************** */
+
 /*
  * jpeglib.h
  *
@@ -887,7 +892,7 @@ typedef JMETHOD(boolean, jpeg_marker_parser_method, (j_decompress_ptr cinfo));
  * characters, you are out of luck.  Get a better compiler.)
  */
 
-#ifdef NEED_SHORT_EXTERNAL_NAMES
+/* #ifdef NEED_SHORT_EXTERNAL_NAMES
 #define jpeg_std_error		jStdError
 #define jpeg_CreateCompress	jCreaCompress
 #define jpeg_CreateDecompress	jCreaDecompress
@@ -941,7 +946,15 @@ typedef JMETHOD(boolean, jpeg_marker_parser_method, (j_decompress_ptr cinfo));
 #define jpeg_abort		jAbort
 #define jpeg_destroy		jDestroy
 #define jpeg_resync_to_restart	jResyncRestart
-#endif /* NEED_SHORT_EXTERNAL_NAMES */
+#endif */ /* NEED_SHORT_EXTERNAL_NAMES */
+
+
+/* FLTK: enable prefixes
+ *
+ * Generate header with the following command:
+ * nm --extern-only --defined-only libfltk_jpeg.a | awk '{print $3}' | sort | uniq | sed '/^$/d' | awk '{print "#define "$1" fltk_"$1}' > jpeg_prefixes.h
+ */
+#include "jpeg_prefixes.h"
 
 
 /* Default error-management setup */
